GOVERNANCE.md

```diff
@@ -0,0 +1,114 @@
+- **Documentation** - Website and Documentation team
+- **Packages** - Works on our core packages
+
+### Discord
+
+Our Discord server is the preferred communication method for team conversations and decision-making.
+We define a number of official roles on here (as well as community ones) to make sure that administration and moderation requirements are met.
+
+#### Admins
+
+Responsible for overall configuration and maintenance of the server.
+- Creating, editing, removing channels
+- Administrating roles and members
+- Maintaining bot and app integrations
+- Kicking/Banning users that break our rules and guidelines and code of conduct.
+
+#### Moderators
+
+Responsible for the day to day moderation of the server and community interactions.
+- Ensuring civil discourse is taking place and taking action (warning users etc.)
+- De-escalating any situations or heated conversations
+
+#### Community roles
+
+Community roles can be self-granted via the `#role-select` channel.
+These members are people who are interested or participate within the Pulsar community and are expected to adhere to our code of conduct but otherwise do not have any official or specific responsibilities.
+These roles are designed to help people get updates and communicate within the specific areas of the project that they are interested in.
+Pulsar team members will also be part of these groups but they may not entirely correlate to the memberships and responsibilities granted in the GitHub sub-teams.
+These roles are:
+
+##### Core teams
+- Core - works on the Main part of our editor
+- Backend - API and server Development
+- Documentation - Website and Documentation team
+- Packages - Works on our core packages
+- Developer - Just fills in anywhere that they want
+
+##### Operating Systems
+- Linux
+- MacOS
+- Windows
+
+##### Other roles
+- Community Packages - join if you're into making packages for the editor
+- Updates - Join this if you're interested in receiving updates as they come out
+
+## Team member & role additions/removals
+
+Team/role administration and membership decision making is mostly covered by our [Policies](https://github.com/pulsar-edit/.github/blob/main/POLICY.md) but this section details the criteria and justifications for adding or removing team members and/or roles.
+
+### Adding members/roles
+
+Adding team members is performed in much the same way as any other decision (see [Policies](https://github.com/pulsar-edit/.github/blob/main/POLICY.md)) by proposals being made to add somebody to a team or role and a vote then taking place on that proposal.
+If the vote passes without any significant disagreements then the member can be added to the relevant team or role.
+There is no strict set of criteria for addition but generally people who are active in the community and project for an extended period of time, have contributed and added value, and share in the organization's goals & vision can be considered.
```

Comment:
I'm not sure we had formalized that adding members goes to a vote, though now that I think about it, it has been done _most_ of the times.

Exceptions were the somewhat haphazard early days when it wasn't clear when/if we were going to branch off from atom-community, and we needed some people with permissions just to have people with permissions and to get the ball rolling on Pulsar (not just PRs and discussions at atom-community org).

But this does feel like putting it in writing, which it is. Just noting it. Mixed feelings, since feels _mildly_ "documentation before policy" "cart before horse", but I'm leaning toward this _does_ feel like it is de-facto policy and I seem to recall informal agreement on this one. Just this makes it somewhat formal.

Reply:
Sorry it took so long to come back to this, not sure what happened.
So yeah, this one I think has indeed been *de-facto* policy, the intial grouping was one thing but this has been the process since I joined (02/08) at the very least, including raising access levels.